src/main/java/com/notably/model/block/BlockTreeItemImpl.java

```diff
@@ -0,0 +1,133 @@
+package com.notably.model.block;
+
+import static java.util.Objects.hash;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import com.notably.model.block.exceptions.DuplicateBlockException;
+import com.notably.model.block.exceptions.NoSuchBlockException;
+
+import javafx.collections.ObservableList;
+import javafx.scene.control.TreeItem;
+
+/**
+ * Represents a BlockTreeItem in the Notably data structure.
+ */
+public class BlockTreeItemImpl extends TreeItem<Block> implements BlockTreeItem {
```

Comment:
Hey @firzanarmani ! Don't you think it's quite conflicting that we're making `BlockTreeItemImpl` implements `BlockTreeItem` when it already extends `TreeItem<Block>`?

The operations exposed by `BlockTreeItem` are somewhat similar to those provided by `TreeItem<Block>`. To list them out:
1. `getTreeItem` is not really useful, since it's just returning a reference of the current object.
2. `getBlock` is equivalent to `getValue`
3. `getBlockParent` is equivalent to `getParent`
4. `getObservableChildren` is equivalent to `getChildren`

Reply:
After thinking more about it, I think we should go down the composition road. That is, we make `BlockTreeItemImpl` implement only the `BlockTreeItem` interface and make it contain a `TreeItem<Block>` entry inside of it (but don't let the user of your API know about this implementation detail regarding `TreeItem<Block>`). This way, we can ensure that when in the future we would like to move away from the `TreeItem<Block>` way, we can do without any changes in API. Additionally, doing composition this way enables us to name the methods in `BlockTreeItemImpl` more fluently. For example, we can have the `getBlockChildren` method without any confusion with the `getChildren` method (which in this case is inherited from `TreeItem<Block>`)